db/osqlsession.c

```diff
@@ -693,8 +697,11 @@ static int osql_sess_set_terminate(osql_sess_t *sess)
 {
     int rc = 0;
     sess->terminate = OSQL_TERMINATE;
-    rc = osql_repository_rem(sess, 0, __func__, NULL,
-                             __LINE__); /* already have exclusive lock */
+    Pthread_mutex_unlock(&sess->completed_lock);
+    Pthread_mutex_unlock(&sess->mtx);
+
```

Comment:
This is the real bug, these two should be unlocked before calling osql_repository_rem.